### PR TITLE
URL-encoding in forwarding and ScriptLoadBalancer metrics

### PR DESCRIPTION
**Pull request by mprimi**
- Avoid URL-encoding when forwarding from one node to another.
 - Fix incorrect ScriptLoadBalancer metrics

### CODE REVIEW

**Comment by mprimi** [comment]
@tgianos one more commit to refactor and test `ScriptLoadBalancer`

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java [hunk 1]
```diff
@@ -155,12 +157,72 @@ public Cluster selectCluster(
...
+
+            if (clusterId == null) {
+                tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_NO_PREFERENCE));
+                log.debug("Script returned null, no preference");
+                return null;
+            }
+
+            for (final Cluster cluster : clusters) {
+                if (clusterId.equals(cluster.getId())) {
+                    tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_FOUND));
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Maybe we should tag with the cluster id selected as well?

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java [hunk 2]
```diff
@@ -155,12 +157,72 @@ public Cluster selectCluster(
...
+    static class Evaluator {
+
+        private final ObjectMapper mapper;
+        private final AsyncTaskExecutor asyncTaskExecutor;
+        private final long timeoutLength;
+
+        Evaluator(
+            final ObjectMapper mapper,
+            final AsyncTaskExecutor asyncTaskExecutor,
+            final long timeoutLength) {
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: usually this `) {` is on the next line

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java [hunk 3]
```diff
@@ -155,12 +157,72 @@ public Cluster selectCluster(
...
+            return null; // Should throw
+
+        } finally {
+            this.registry
+                .timer(SELECT_TIMER_NAME, tags)
+                .record(System.nanoTime() - selectStart, TimeUnit.NANOSECONDS);
+        }
+    }
+
+    static class Evaluator {
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: This makes it package private which would force anyone trying to extend it to put that code in this package. Might want to make it explicitly `protected`?

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java [hunk 4]
```diff
@@ -174,139 +236,139 @@ public Cluster selectCluster(
...
-                )
+        private final Environment environment;
+        private final MeterRegistry registry;
+        private final GenieFileTransferService fileTransferService;
+
+
+        Loader(
+            final TaskScheduler taskScheduler,
+            final Environment environment,
+            final GenieFileTransferService fileTransferService,
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Not to increase the scope creep here but we'll probably want to get rid of this service soon so if you were so inclined to flip this to use the `ResourceLoader` I wouldn't object

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java [hunk 5]
```diff
@@ -174,139 +236,139 @@ public Cluster selectCluster(
...
-            );
-            this.isConfigured.set(false);
-        } finally {
-            this.isUpdating.set(false);
-            this.registry
-                .timer(UPDATE_TIMER_NAME, tags)
-                .record(System.nanoTime() - updateStart, TimeUnit.NANOSECONDS);
-            log.debug("Refresh completed");
+        @Nullable
+        CompiledScript get() {
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Is this one of those `@VisibleForTesting` things? Seems unnecessary

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java (hunk 1 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Would be nice to have.
Tho then we should also do the same in the class that uses this component (`JobSpecService`), since more than one `LoadBalancer` may be used.

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java (hunk 5 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: It's part of the script `Loader` interface so it can be mocked easily.
But yeah, package private is probably not the best choice, i'll fix it.

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java (hunk 1 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Added a commit that adds metrics tags

File: genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java (hunk 3 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java]: Fixed, thank you.
